### PR TITLE
Dungeon parsing cleanup and fix for broken multiplier

### PR DESCRIPTION
**Pull request by rohilthopu**
I added in the following

1. parsing for modifiers by their types after splitting
2. added in your hack for fixed teams
3. added a messages var for dmsg/smsg items
4. added enhanced type/attribute variables
5. added score as test_score under the assumption that it might be in correct. as a result, I left your hack in from before. I can't say I would've thought to use bitwise operators for this. 
6. added a section for remaining_modifiers as a debug for parsing the last remaining ones that i have no idea what they mean
7. I left in your other_fields section
8. NEW: Moved the while loop into dungeon_parse to keep the Dungeon file cleaner.

I've tested on my end and it seems like I filled in most of the blanks. I left in try/except statements in case things break so we have a log.

If you can verify that things here are good on your end, I'll clean up and remove the commented out code (or you can whatever works).

EDIT2: Included the broken multiplier fix in this. the calculator is fine, but there was an index out of bounds because len(other_fields) was 0

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,115 +0,0 @@
-# Byte-compiled / optimized / DLL files
```
Comment (nachoapps) [inline .gitignore]: Please back out this change.

Comment (nachoapps) [reply .gitignore]: You also have pyc files in this commit, please back those out too.

File: pad_api_data/pad_etl/common/dungeon_maps.py [hunk 2]
```diff
@@ -37,3 +37,32 @@
     1792: 'Awoken Skills Invalid',
     2048: 'Assists Invalid'
 }
+
+
+btypeChart = {
```
Comment (nachoapps) [inline pad_api_data/pad_etl/common/dungeon_maps.py]: Constants like this should be ALL_CAPS

Comment (nachoapps) [reply pad_api_data/pad_etl/common/dungeon_maps.py]: Also please leave a comment indicating what the key is here.

File: pad_api_data/pad_etl/common/dungeon_maps.py [hunk 3]
```diff
@@ -37,3 +37,32 @@
...
+    64: 'Attacker',
+    68: 'Physical & Attacker',
+    128: 'Devil',
+    132: 'Physical & Devil',
+    192: 'Attacker & Devil Enhanced',
+    256: 'Machine Enhanced',
+    320: 'Attacker & Machine'
+}
+
+battrChart = {
```
Comment (nachoapps) [inline pad_api_data/pad_etl/common/dungeon_maps.py]: Looks like a bitmap key, please leave a note indicating this and the source field, as well as style change.

File: pad_api_data/pad_etl/common/dungeon_parse.py [hunk 4]
```diff
@@ -1,12 +1,21 @@
-# A small object to hold floor modifier information
+from dungeon_maps import btypeChart, battrChart
+
+
 class Modifier:
     def __init__(self):
         self.requiredDungeon = 0
-        self.modifiers = []
+        self.remainingModifiers = []
```
Comment (nachoapps) [inline pad_api_data/pad_etl/common/dungeon_parse.py]: You should familiarize yourself with the PEP8 style guide:
https://www.python.org/dev/peps/pep-0008/#naming-conventions

In general your code tends not to conform.

File: pad_api_data/pad_etl/common/dungeon_parse.py [hunk 5]
```diff
@@ -1,12 +1,21 @@
...
         self.entryRequirement = "None"
+        self.modifiers = {}
+        self.messages = []
+        self.fixedTeam = {}
+        self.enhancedType = "None"
+        self.enhancedAttribute = "None"
```
Comment (nachoapps) [inline pad_api_data/pad_etl/common/dungeon_parse.py]: Single quotes are preferred; regardless the style should be consistent in the same file.

File: pad_api_data/pad_etl/common/dungeon_parse.py [hunk 6]
```diff
@@ -15,18 +24,41 @@ def getLast(raw):
...
         return modifiers
 
+    elif val == 8:
+        modifiers.score = raw[pos + 1]
+        return modifiers
+
     elif val == 32:
-        modifiers.entryRequirement = parse32[int(raw[pos + 1])](raw)
+        modifiers.messages.append(parse32[int(raw[pos + 1])](raw))
+        # modifiers.messages.append(parse32[int(raw[pos + 2])](raw))
```
Comment (nachoapps) [inline pad_api_data/pad_etl/common/dungeon_parse.py]: Leave a note explaining why this is disabled.

File: pad_api_data/pad_etl/common/pad_util.py [hunk 7]
```diff
@@ -143,467 +143,468 @@ def __init__(self):
...
-        multipliers.hp *= get_last(other_fields)
-
-    elif skill in [24, 49, 149]:
-        multipliers.rcv *= get_last(other_fields)
-
-    # RCV and ATK
-    elif skill in [28, 64, 75, 79, 103]:
-        multipliers.atk *= get_last(other_fields)
-        multipliers.rcv *= get_last(other_fields)
+    if length > 0:
```
Comment (nachoapps) [inline pad_api_data/pad_etl/common/pad_util.py]: The change in this file is waaaaay too big to review cleanly.

In general, if you're refactoring / cleaning up that should be in it's own CL, separate from changes to fix bugs / add features.

File: pad_api_data/pad_etl/data/dungeon.py [hunk 8]
```diff
@@ -30,88 +30,80 @@ def __init__(self, raw: List[Any]):
...
-        self.requiredDungeon = modifiers.requiredDungeon
-
-        self.modifiers = modifiers.modifiers
+        # These need to be parsed depending on flags
+        self.other_modifier = raw7_map[int(raw[7])]
+
+        try:
+            modifiers = getModifiers(raw)
+        except Exception as e:
+            print("Error:", e, "on parse of values", int(raw[8:]))
```
Comment (nachoapps) [inline pad_api_data/pad_etl/data/dungeon.py]: Does this ever happen on a parse of the current dungeon file?  If it does, then this change shouldn't be committed.

Also this is probably the wrong place to swallow this exception. I'd rather it propagate up and the caller decide what to do.

File: pad_api_data/pad_etl/data/dungeon.py [hunk 9]
```diff
@@ -30,88 +30,80 @@ def __init__(self, raw: List[Any]):
...
-                    'rcv_plus': details[3] if full_record else 0,
-                    'awakening_count': details[4] if full_record else 0,
-                    'skill_level': details[5] if full_record else 0,
-                }
+        self.remaining_fields = raw[pos + 1:]
+        #
+        # # Modifiers parsing doesn't seem to always work
+        # # Hacked up version for dungeon modifiers, needed for
+        # # enemy parsing.
+        # self.modifiers_clean = {
```
Comment (nachoapps) [inline pad_api_data/pad_etl/data/dungeon.py]: This field is currently used elsewhere and it doesn't look like you've replaced it or changed the usage. 

Committing this will break things, which is not good.

File: pad_api_data/pad_etl/data/dungeon.py [hunk 10]
```diff
@@ -30,88 +30,80 @@ def __init__(self, raw: List[Any]):
...
+        #     'hp': 1.0,
+        #     'atk': 1.0,
+        #     'def': 1.0,
```
Comment (nachoapps) [inline pad_api_data/pad_etl/data/dungeon.py]: There's no reason to leave all this code in here commented out; we can always get it out of history if we need to look at it.

Especially if you've tied out your changes against the temporary code I swapped in here. Have you done that?

File: pad_api_data/pad_etl/data/dungeon.py [hunk 11]
```diff
@@ -30,88 +30,80 @@ def __init__(self, raw: List[Any]):
...
+        self.entry_requirement = modifiers.entryRequirement
+        self.required_dungeon = modifiers.requiredDungeon
+        self.remaining_modifiers = modifiers.remainingModifiers
+        self.stat_modifiers = modifiers.modifiers
+        self.enhanced_type = modifiers.enhancedType
+        self.enhanced_attribute = modifiers.enhancedAttribute
+        self.messages = modifiers.messages
+        self.fixed_team = modifiers.fixedTeam
+        self.remaining_modifiers = modifiers.remainingModifiers
+        self.test_score = modifiers.score
```
Comment (nachoapps) [inline pad_api_data/pad_etl/data/dungeon.py]: test_score?

File: pad_api_data/pad_etl/data/dungeon.py [hunk 12]
```diff
@@ -121,27 +113,27 @@ def __init__(self, raw: List[Any]):
...
             #  dungeonId: Number(self.remaining_fields[i++]),
             #  floorId: Number(self.remaining_fields[i++])
-            #};
+            # };
         if ((self.flags & 0x4) != 0):
             i += 1
-            #self.beginTime = fromPADTime(self.remaining_fields[i++]);
+            # self.beginTime = fromPADTime(self.remaining_fields[i++]);
         if ((self.flags & 0x8) != 0):
             self.score = int(self.remaining_fields[i]);
```
Comment (nachoapps) [inline pad_api_data/pad_etl/data/dungeon.py]: Are you not confident in your score parsing? Why is this still here.

File: pad_api_data/pad_etl/data/skill.py [hunk 13]
```diff
@@ -54,9 +54,9 @@ def __init__(self, skill_id: int, raw: List[Any]):
 
         # NEW FIELDS. The skills that a skill links to if it has multiple
         # clauses/conditions for activation
-        self.skill_part_1_id = None
-        self.skill_part_2_id = None
-        self.skill_part_3_id = None
+        self.skill_part_1_id = -1
```
Comment (nachoapps) [inline pad_api_data/pad_etl/data/skill.py]: Why this change? None is a better choice.

**Comment by rohilthopu** [comment]
I'll close this for now, but clean up big time and resubmit properly.
